File: vaultExport.js
const { GoogleAuth } = require('google-auth-library');
const axios = require('axios');
const fs = require('fs');

async function run() {
  const key = JSON.parse(process.env.GOOGLE_SERVICE_ACCOUNT_JSON);

  const auth = new GoogleAuth({
    credentials: key,
    scopes: [
      'https://www.googleapis.com/auth/ediscovery',
      'https://www.googleapis.com/auth/devstorage.read_only'
    ],
    clientOptions: {
      subject: process.env.WORKSPACE_ADMIN_EMAIL
    }
  });

  const client = await auth.getClient();
  const token = await client.getAccessToken();

  const matterId = process.env.VAULT_MATTER_ID;

  const res = await axios.post(
    `https://vault.googleapis.com/v1/matters/${matterId}/exports`,
    {
      name: `voice-export-${new Date().toISOString()}`,
      query: {
        corpus: 'VOICE',
        dataScope: 'ALL_DATA',
        accountInfo: {
          emails: [process.env.VOICE_USER_EMAIL]
        }
      },
      exportOptions: {
        voiceOptions: { exportFormat: 'MBOX' }
      }
    },
    {
      headers: {
        Authorization: `Bearer ${token.token}`
      }
    }
  );

  console.log('Export created:', res.data.id);
}

run().catch(err => {
  console.error(err.response?.data || err.message);
  process.exit(1);
});

File: vaultExport.py
import os
from pathlib import Path
import requests
import shutil
import google.auth.transport.requests as google_requests
from utils import build_filename, upload_to_drive, extract_zip_files, get_auth_credentials, get_audio_files

TEMP_DIR = "./temp"
EXTRACT_DIR = "./temp/extracted"


def download_zip_files(gcs_url, credentials):
    print(f"Downloading: {gcs_url}")
    headers = {"Authorization": f"Bearer {credentials.token}"}
    zip_path = os.path.join(TEMP_DIR, "export.zip")

    with requests.get(gcs_url, headers=headers, stream=True, timeout=30) as response:
        response.raise_for_status()
        with open(zip_path, 'wb') as f:
            shutil.copyfileobj(response.raw, f)

    file_size = os.path.getsize(zip_path)
    print(f"Download completed. File size: {file_size} bytes")



def download_and_upload(completed_export, credentials):
    
    files = completed_export.get('cloudStorageSink', {}).get('files', [])
    if not files:
        print("No files found in the completed export")
        return
    
    for file_info in files:
        gcs_url = f"https://storage.googleapis.com/{file_info['bucketName']}/{file_info['objectName']}"
        download_zip_files(gcs_url, credentials)

    zip_path = os.path.join(TEMP_DIR, "export.zip")
    extract_zip_files(zip_path)

    audio_files = get_audio_files()
    print(f"All files found: {audio_files}")
    
    for file_path in audio_files:
        print(f"Uploading file: {file_path}")
        full_path = os.path.join(EXTRACT_DIR, file_path)
        file_name = build_filename(os.path.basename(file_path))

        upload_to_drive(credentials, full_path, file_name)

        print("All recordings uploaded")



def run():
    """Main execution function"""
    # Create directories
    Path(EXTRACT_DIR).mkdir(parents=True, exist_ok=True)

    credentials = get_auth_credentials()

    # Impersonate workspace admin if specified
    workspace_admin_email = os.environ.get('WORKSPACE_ADMIN_EMAIL')
    if workspace_admin_email:
        credentials = credentials.with_subject(workspace_admin_email)

    # --- Get latest completed export ---
    vault_matter_id = os.environ.get('VAULT_MATTER_ID')
    if not vault_matter_id:
        raise ValueError("VAULT_MATTER_ID environment variable not set")

    # Get access token
    request = google_requests.Request()
    credentials.refresh(request)

    exports_url = f"https://vault.googleapis.com/v1/matters/{vault_matter_id}/exports"
    headers = {"Authorization": f"Bearer {credentials.token}"}

    response = requests.get(exports_url, headers=headers)
    response.raise_for_status()
    exports_data = response.json()

    completed_exports = []
    for export in exports_data.get('exports', []):
        if export.get('status') == 'COMPLETED':
            completed_exports.append(export)

    if not completed_exports:
        print("No completed export found")
        return

    for completed_export in completed_exports:
        download_and_upload(completed_export, credentials)

if __name__ == "__main__":
    try:
        run()
    except Exception as err:
        print(f"Error: {err}")
        exit(1)